### PR TITLE
move GitInfo into lib / test / scope / lazy it

### PR DESCRIPTION
**Pull request by grosser**
@sandlerr @omgbob 

https://github.com/zendesk/samson/pull/1378

### CODE REVIEW

File: test/lib/samson/git_info_test.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+require_relative '../../test_helper'
+
+SingleCov.covered!
+
+describe Samson::GitInfo do
+  describe ".version" do
+    before { Samson::GitInfo.instance_variable_set(:@version, nil) }
+
+    it "works" do
+      Samson::GitInfo.version.to_s.must_match /^\d.\d.\d$/
```
Comment (omgbob) [inline test/lib/samson/git_info_test.rb]: Maybe `/^\d(.\d){2,3}$/` as the version can be 3 or 4 parts (ie `1.8.1` or `1.8.5.6`)

**Comment by grosser** [comment]
4 part versions work too ...

Gem::Version.new('1.2.3.4') > Gem::Version.new('1.2.3.3')
=> true
>> Gem::Version.new('1.2.3.4') > Gem::Version.new('1.2.3.4')
=> false

File: test/lib/samson/git_info_test.rb (hunk 1 above)
Comment (grosser) [inline test/lib/samson/git_info_test.rb]: ah ... did not look at travis :D

Comment (grosser) [inline test/lib/samson/git_info_test.rb]: fixed ... good to go ?

File: test/lib/samson/git_info_test.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+require_relative '../../test_helper'
+
+SingleCov.covered!
+
+describe Samson::GitInfo do
+  describe ".version" do
+    before { Samson::GitInfo.instance_variable_set(:@version, nil) }
+
+    it "works" do
+      Samson::GitInfo.version.to_s.must_match /^\d{\.\d}{2,3}/
```
Comment (omgbob) [inline test/lib/samson/git_info_test.rb]: Should be parenthesis for the first brackets, not curly braces:

```ruby
/^\d(\.\d){2,3}/

Comment (grosser) [inline test/lib/samson/git_info_test.rb]: somehow that passed ... I guess missing $ :)

fixed!

**Pull request merged**